tests/e2e/fixtures/cognito_dependencies.py

```diff
@@ -94,7 +95,11 @@ def on_create():
 
     def on_delete():
         cfg = metadata.get("cognito_dependencies") or cognito_deps
-        delete_cognito_dependency_resources(cfg)
+        print("keep Success Option for Cognito Fixture:")
+        print(keep_successfully_created_resource(request))
+        if keep_successfully_created_resource(request) == False:
+            print("start to uninstall Cognito Dependencies...")
+            delete_cognito_dependency_resources(cfg)
```

Comment:
this logic is not necessary, the keep success condition is already handled by `configure_resource_fixture()`

```
def configure_resource_fixture(
    metadata, request, resource_details, metadata_key, on_create, on_delete
):
    """
    Helper method to create resources if required and configure them for teardown.

    If the resource is not present in the metadata a new resource will be created and added to
    the metadata.

    If a resource is present in the metadata it will not be created.

    If the `--keepsuccess` flag is specified, successfully created resources (e.g. those that
    did not raise an exception) will not be deleted.
    """
    successful_creation = False

    def delete():
        if successful_creation and keep_successfully_created_resource(request):
            return
        on_delete()
```